Week2/ex2.py

```diff
@@ -0,0 +1,21 @@
+import functools
+
+class InvalidTypeError(TypeError):
+    pass
+
+def type_checker(arg_type):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(arg):
+            if not isinstance(arg, arg_type):
+                raise InvalidTypeError(f"Expected {arg_type.__name__} but got {type(arg).__name__}")
+            return func(arg)
+        return wrapper
+    return decorator
+
+@type_checker(int)
+def square(x):
+    return x ** 2
+
+#print(square(5)) # Output: 25
```

Comment:
You should not leave documented code in your work.
Better put it in a test function and call it from __name__ == "__main__"